### PR TITLE
2-3 additional tests.

### PR DESCRIPTION
**Pull request by lemire**
No big deal, just want to make sure we expose problems later if we change percent/space encoding.

### CODE REVIEW

File: tests/basic_tests.cpp [hunk 1]
```diff
@@ -20,6 +20,34 @@ TYPED_TEST(basic_tests, insane_url) {
...
+  r = ada::parse<TypeParam>("http://www.google%X%.com/");
+  ASSERT_FALSE(r);
+  SUCCEED();
+}
+
+TYPED_TEST(basic_tests, spaces_spaces) {
+  auto r = ada::parse<TypeParam>("http://www.google.com/%37/ /");
+  ASSERT_TRUE(r);
+  ASSERT_EQ(r->get_href(), "http://www.google.com/%37/%20/");
+  r = ada::parse<TypeParam>("http://www.google com/");
```
Comment (anonrig) [inline tests/basic_tests.cpp:36]: Should we also do a test where `r->set_href()` on all of these tests?

**Review by anonrig — APPROVED** [review]

File: tests/basic_tests.cpp (hunk 1 above)
Comment (lemire) [reply tests/basic_tests.cpp:36]: Ok.

Comment (lemire) [reply tests/basic_tests.cpp:44]: Done.

**Comment by lemire** [comment]
@anonrig I am using this PR to make sure that our main source files are pure ASCII. This is a good default because some compilers like Visual Studio do not default on UTF-8. So far this hasn't been a problem, but it could become one for some users.

ASCII is safe.

**Review by anonrig — APPROVED** [review]

**Review by anonrig — APPROVED** [review]

**Pull request merged**